wazo_confd/plugins/user/middleware.py

```diff
@@ -1,22 +1,103 @@
 # Copyright 2022 The Wazo Authors  (see the AUTHORS file)
 # SPDX-License-Identifier: GPL-3.0-or-later
 
+from xivo_dao.helpers.db_manager import Session
 from xivo_dao.alchemy.userfeatures import UserFeatures as User
+from xivo_dao.resources.switchboard import dao as switchboard_dao
 
-from .schema import UserSchemaNullable
+from .schema import UserListItemSchema
 
 
 class UserMiddleWare:
-    def __init__(self, service):
+    def __init__(self, service, wazo_user_service, middleware_handle):
         self._service = service
-        self._schema = UserSchemaNullable()
+        self._wazo_user_service = wazo_user_service
+        self._middleware_handle = middleware_handle
+        self._schema = UserListItemSchema()
 
-    def create(self, body, tenant_uuid):
+    def create(self, body, tenant_uuid, tenant_uuids):
         form = self._schema.load(body)
         form['tenant_uuid'] = tenant_uuid
+
+        auth = form.pop('auth', None)
+        lines = form.pop('lines', None) or []
+        incalls = form.pop('incalls', None) or []
+        groups = form.pop('groups', None) or []
+        switchboards = form.pop('switchboards', None) or []
+
         model = User(**form)
         model = self._service.create(model)
-        return self._schema.dump(model)
+        user_dict = self._schema.dump(model)
+        user_dict['lines'] = []
+        user_dict['incalls'] = []
+        user_dict['groups'] = []
+        user_dict['switchboards'] = []
+
+        if auth:
+            auth['uuid'] = user_dict['uuid']
+            auth['tenant_uuid'] = user_dict['tenant_uuid']
+            user_dict['auth'] = self._wazo_user_service.create(auth)
+
+        if lines:
+            for line_body in lines:
+                line = self._middleware_handle.get('line').create(
+                    line_body, tenant_uuid, tenant_uuids
+                )
+                self._middleware_handle.get('user_line_association').associate(
+                    user_dict['id'], line['id'], tenant_uuids
+                )
+                user_dict['lines'].append(line)
+            Session.expire(model)
```

Comment:
Could we use `Session.expire(model, ['lines'])` here instead of expiring everything?